### PR TITLE
DRAFT: Add major changes to the constitution.

### PR DESCRIPTION
**Pull request by tfpk**
The object of this WIP change is to convert the constitution to be based both on Arc's Constitution and NSW's model constitution.

### CODE REVIEW

File: CONSTITUTION.rst [hunk 1]
```diff
@@ -0,0 +1,861 @@
...
+      #. They have applied to a member of the club Executive of CSESoc to become an associate member, and
+      #. They would not cause a violation of the affiliation requirements of Arc, and
+      #. They pay a membership fee, as set by the club Executive, and
+      #. They complete a membership form prepared by the club Executive.
+   #. An associate member whose membership is by application to the Executive shall cease to be a member of CSESoc at the end of the calendar year, unless they would otherwise retain that membership.
+#. Notwithstanding other provisions of this section, a person ceases to be a member of CSESoc if the person:
+
+   #. dies, or
+   #. notifies the Executive in writing to request their membership be terminated, or
+   #. is removed from the Executive under the provisions described in `Disciplining of Members`_.
```
Comment (siddhantvirmani) [inline CONSTITUTION.rst:149]: Dead link - should this link to Impeachment instead?

Comment (tfpk) [reply CONSTITUTION.rst:149]: I think a few of the links are broken. I'll check them out, thanks :+1:

File: CONSTITUTION.rst [hunk 2]
```diff
@@ -0,0 +1,861 @@
...
+
+#. The society shall publicly maintain a Grievance Resolution Policy and Procedure.
+
+   #. This Policy and Procedure shall initially be the Grievance Resolution Policy and Procedure of the Unincorporated Association which incorporates under this constitution.
+   #. This Policy and Procedure may be modified by a vote of the Executive.
+
+#. A dispute between a member and another member (in their capacity as members) of the association shall be referred to the CSESoc Grievance officer; who shall act in accordance with the Grievance Resolution Policy and Procedure.
+#. A dispute between a member or members and CSESoc, is to be referred to a Community Justice Centre for mediation under the Community Justice Centres Act 1983.
+#. If a dispute is not resolved by mediation within 3 months of the referral to a Community Justice Centre, the dispute is to be referred to arbitration.
+#. The Commercial Arbitration Act 2010 applies to a dispute referred to arbitration.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:199]: This is just the default given in the Model Constitution. I'm pretty sure you can use a different dispute resolution policy if you want (i.e. you don't have to use their exact procedure).

File: CONSTITUTION.rst [hunk 3]
```diff
@@ -0,0 +1,861 @@
...
+#. *Arc* shall mean Arc @ UNSW Limited; 
+#. *Tharunka* and *Blitz* shall mean the official publications of Arc; 
+#. *Re-affiliation* shall mean re-affiliation with Arc; 
+#. *Full members* shall mean full members of the Computer Science and Engineering Society of UNSW; 
+#. *Associate members* shall mean associate members of the Computer Science and Engineering Society of UNSW; 
+#. *The Executive* shall mean the Executive of the Computer Science and Engineering Society of UNSW; 
+#. *A Portfolio* shall mean a group of members of the society, selected by The Executive to perform a particular task or tasks, as delegated to by The Executive.
+#. *The Annual General Meeting* shall mean the Annual General Meeting of the CSE Society; 
+#. An *academic day* shall mean a day during the first, second or third term of the University’s 
+   cademic year which is not a Saturday, Sunday, Public Holiday or University Holiday;
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:90]:
```suggestion
  academic year which is not a Saturday, Sunday, Public Holiday or University Holiday;
```

File: CONSTITUTION.rst [hunk 4]
```diff
@@ -0,0 +1,861 @@
...
+---------------
+
+    Notes from the Model Constitution:
+    1. Schedule 1 to the Act provides that an association’s constitution is to address CSESoc’s financial year.
+    2. Clause 19 of the Regulation contains a substitute clause 47 for certain associations incorporated under CSESocs Incorporation Act 1984.
+	
+#. The financial year of CSESoc is:
+
+   #. the period of time commencing on the date of incorporation of CSESoc and ending on the following 31 December, and
+   #. each period of 12 months after the expiration of the previous financial year of CSESoc, commencing on 1 July and ending on the following 30 June.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:861]: This line and the one immediately above are in conflict. The ending date needs to be the same in both clauses.

File: CONSTITUTION.rst [hunk 5]
```diff
@@ -0,0 +1,861 @@
...
+#. An annual general meeting must be specified as that type of meeting in the notice convening it.
+
+Extraordinary general meetings - calling of
+-------------------------------------
+
+    Copied from section 27 of the Model Constitution, but with extra stuff from Section 5 of the CSESoc constitution.
+
+#. The Executive may, whenever it thinks fit, convene a special general meeting of CSESoc.
+#. Extraordinary General Meetings must be held on an academic day.
+#. Each full member is entitled to one vote.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:601]: Seems to be redundant with the "Voting" section.

File: CONSTITUTION.rst [hunk 6]
```diff
@@ -0,0 +1,861 @@
...
+#. A right, privilege or obligation which a person has by reason of being a member of CSESoc:
+
+   #. is not capable of being transferred or transmitted to another person, and
+   #. terminates on cessation of the person’s membership.
+
+Register of Members
+-------------------
+
+
+#. The secretary must establish and maintain a register of members of CSESoc (whether in written or electronic form) specifying the name and postal, residential or email address of each person who is a member of CSESoc together with the date on which the person became a member.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:165]: Nothing that needs to be changed _per se_, but be aware that the law requires extra records to be kept that aren't listed in the Model Constitution. The following records must be kept about the Executive:
* name, date of birth, residential address
* dates they enter/leave the Executive
* names and start/end dates of anyone in the positions of "president, vice-president (N/A), secretary or treasurer"

File: CONSTITUTION.rst [hunk 7]
```diff
@@ -0,0 +1,861 @@
...
+
+    CSESoc Consitution, Section 1
+    Model Constitution, Section 1
+
+1. In this constitution:
+
+   #. *secretary* means:
+
+      #. the person holding office under this constitution as secretary of CSESoc, or
+      #. if no person holds that office - the public officer of CSESoc.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:74]: I would suggest putting somewhere in the constitution who is/can be the public officer.

File: CONSTITUTION.rst [hunk 8]
```diff
@@ -0,0 +1,861 @@
...
+#. *Associate members* shall mean associate members of the Computer Science and Engineering Society of UNSW; 
+#. *The Executive* shall mean the Executive of the Computer Science and Engineering Society of UNSW; 
+#. *A Portfolio* shall mean a group of members of the society, selected by The Executive to perform a particular task or tasks, as delegated to by The Executive.
+#. *The Annual General Meeting* shall mean the Annual General Meeting of the CSE Society; 
+#. An *academic day* shall mean a day during the first, second or third term of the University’s 
+   cademic year which is not a Saturday, Sunday, Public Holiday or University Holiday; 
+#. A *business* day shall mean a day which is not a Saturday, Sunday, Public Holiday, 
+   niversity Holiday or within a University exam period; 
+#. The *School of CSE* or any references to the acronym *CSE* shall mean Computer 
+   cience and Engineering; and
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:94]:
```suggestion
   Science and Engineering; and 
```

File: CONSTITUTION.rst [hunk 9]
```diff
@@ -0,0 +1,861 @@
...
+#. *A Portfolio* shall mean a group of members of the society, selected by The Executive to perform a particular task or tasks, as delegated to by The Executive.
+#. *The Annual General Meeting* shall mean the Annual General Meeting of the CSE Society; 
+#. An *academic day* shall mean a day during the first, second or third term of the University’s 
+   cademic year which is not a Saturday, Sunday, Public Holiday or University Holiday; 
+#. A *business* day shall mean a day which is not a Saturday, Sunday, Public Holiday, 
+   niversity Holiday or within a University exam period; 
+#. The *School of CSE* or any references to the acronym *CSE* shall mean Computer 
+   cience and Engineering; and 
+#. *Subjects* shall mean units of study offered by the University in progression to the 
+   ward of a degree.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:96]: This line appears to be unused.

File: CONSTITUTION.rst [hunk 10]
```diff
@@ -0,0 +1,861 @@
...
+#. On dissolution of the society, the society is not to distribute assets to members. All assets are to be distributed to an organisation with similar goals or objectives that also prohibits the distribution of assets to members. This organisation may be nominated at the dissolution meeting of the society. If no other legitimate organisation is nominated, Arc will begin procedures to recover any property, monies or records belonging to the society which it perceives would be useful to other Arc-affiliated clubs. 
+        
+Distribution of property on winding up of association 
+---------------
+
+    Copied from section 42 of the Model Constitution.
+    Note from the model constituion:
+    Section 65 of the Act provides for distribution of surplus property on the winding up of an association.
+
+#. Subject to the Act and the Regulations, in a winding up of CSESoc, any surplus property of CSESoc is to be transferred to another organisation affiliated with Arc, with similar objects and which is not carried on for the profit or gain of its individual members.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:795]: This clause and the one above should be merged together; seems redundant.

**Review by Qantas94Heavy — COMMENTED** [review]
Found a few typos and a couple of potential conflicts between association legislation and Arc policy -- maybe think about talking to Arc about that.

P.S. I hope you guys don't have more than $250k revenue in a year :wink:

File: CONSTITUTION.rst [hunk 11]
```diff
@@ -0,0 +1,861 @@
...
+
+   #. Should the society have less than 15 full members, half of the number of full members shall constitute a quorum.
+
+#. If within half an hour after the appointed time for the commencement of a general meeting a quorum is not present, the meeting:
+
+   #. if convened on the requisition of members—is to be dissolved, and
+   #. in any other case—is to stand adjourned to the same day in the following week at the same time and (unless another place is specified at the time of the adjournment by the person presiding at the meeting or communicated by written notice to members given before the day to which the meeting is adjourned) at the same place.
+
+
+#. If at the adjourned meeting a quorum is not present within half an hour after the time appointed for the commencement of the meeting, the members present (being at least 3) are to constitute a quorum.
```
[the remark this answers is not in the document]
Comment (MrSpanishPear) [reply CONSTITUTION.rst:652]: +1 - from memory a  quorum of 15 members or half the society, whatever is lesser

File: CONSTITUTION.rst [hunk 12]
```diff
@@ -0,0 +1,861 @@
...
+    The CSESoc Constitution allows them, so we do too.
+
+#. General meeting requirements for all general meetings are as follows:
+
+   #. All voting at general meetings shall be with a simple majority required for a resolution to be passed;
+   #. Except where specified otherwise, each member (full, or associate) shall be allowed 1 vote.
+   #. Proxies shall be allowed in meetings and the procedure shall comply with the requirements of Arc;
+   #. In the case of equality of voting, the motion shall be defeated. 
+   #. Constitutional changes must be in the form of a motion on notice to an Annual or Extraordinary General Meeting; and 
+   #. Constitutional changes passed at an Annual or Extraordinary General Meeting will not take effect until ratified by Arc.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:549]:
```suggestion
   #. Constitutional changes passed at an Annual or Extraordinary General Meeting must be approved by Arc for CSESoc to remain affiliated with Arc.
```
This should be updated to the current Arc model constitution wording. As CSESoc would now have its own legal identity, Arc cannot technically prevent constitution changes from taking effect. (Arc affiliation would be suspended though.)

File: CONSTITUTION.rst [hunk 13]
```diff
@@ -0,0 +1,861 @@
...
+    This is copied from section 7 of the CSESoc constitution.
+
+#.  Dissolution of the Computer Science and Engineering Society of UNSW will occur after the following conditions have been met:
+
+   #. An Extraordinary General Meeting is petitioned in writing as set out in 4.11; 
+   #. Procedures for notification as set out in 4.1.3 are followed, and the reasons for the proposed dissolution are included with the notification to Arc; 
+   #. Quorum for the meeting to dissolve the society shall be twenty (20) members or three-quarters of the society membership, whichever is the lesser; 
+   #. No other business may be conducted at the meeting to dissolve the society; 
+   #. After the petitioning body has stated its case any opposition must be given the opportunity to reply, with at least ten minutes set aside for this purpose; 
+   #. A vote is taken and the motion to dissolve lapses if opposed by fifteen (15) or more members of the society; and
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:781]: Note that the dissolution motion must be passed by special resolution, i.e. 21 days' notice and 75% vote is required.

File: CONSTITUTION.rst [hunk 14]
```diff
@@ -0,0 +1,861 @@
...
+#. Associate membership shall be subject to the following conditions:
+
+   #. A person shall be an associate member of the society if:
+
+      #. They are currently enrolled in a course administered by the School of Computer Science and Engineering.
+   #. An associate member whose membership is by virtue of enrollment in a course shall cease to be a member of CSESoc at the beginning of the term after the term of the course they enrolled in, unless they would otherwise retain that membership.
+   #. The executive shall have the power to reasonably interpret which term constitutes the "term after the term of the course they enrolled in".
+   #. A person shall also be an associate member of the society if:
+
+      #. They have applied to a member of the club Executive of CSESoc to become an associate member, and
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:140]: Minor thing, but would recommend removing references to "the club" for consistency's sake.

File: CONSTITUTION.rst [hunk 15]
```diff
@@ -0,0 +1,861 @@
...
+      #. The Executive is petitioned by fifteen (15) members to instigate impeachment proceedings;
+      #. The members of the society are formally notified of the motion upon notice of an Extraordinary General Meeting under Section 4.2;
+      #. The Executive  member concerned is notified of the motion and reasons for termination in writing at least five (5) academic days prior to the Extraordinary General Meeting;
+      #. The Executive member concerned is given five (5) minutes to speak against the motion at the Extraordinary General Meeting; and
+      #. The motion is carried by the Extraordinary General Meeting.
+
+   #. Any vacancy in the society Executive must be filled by following the election procedure under Section 8.
+
+      #. If there is a tie for the vacant executive position, the current executive shall have a casting vote in the election.
+      #. The candidate must accept the offer of Executive position at that meeting, or alternatively at an Extraordinary General Meeting before they may fill the position.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:263]: This whole vacancy bit appears to be somewhat duplicated in the "Casual Vacancies" section way down below.

File: CONSTITUTION.rst [hunk 16]
```diff
@@ -0,0 +1,861 @@
...
+   #. a signature may be transmitted, and a requisition may be lodged, by electronic means.
+
+#. The format, procedures, notice and quorum for an Extraordinary General Meeting shall be the same as for an Annual General Meeting, except that the Executive nominations shall not take place unless specifically notified.
+
+Notice
+------
+
+    Copied from section 28 of the Model Constitution.
+
+#. Except if the nature of the business proposed to be dealt with at a general meeting requires a special resolution of CSESoc, the secretary must, at least 14 days before the date fixed for the holding of the general meeting, give a notice to each member specifying the place, date and time of the meeting and the nature of the business proposed to be transacted at the meeting.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:628]: This section says 14 days' notice is required, but in the _"Annual general meetings - holding of"_ section, it says that only 5 academic days is required.

**Comment by tfpk** [comment]
Okay I think I've resolved all the comments made above.
 - @siddhantvirmani 
 - @Qantas94Heavy 
 - @SpanishPear 

Thanks so much for your contributions, they've been incredibly helpful. Please let me know/make a comment if anything else seems out of order.

**Comment by siddhantvirmani** [comment]
Looks good to me. Great work - and congratulations!

**Comment by SpanishPear** [comment]
LGTM at another glance - :partying_face:

File: CONSTITUTION.rst [hunk 17]
```diff
@@ -0,0 +1,864 @@
...
+---------------
+
+    Notes from the Model Constitution:
+    1. Schedule 1 to the Act provides that an association’s constitution is to address CSESoc’s financial year.
+    2. Clause 19 of the Regulation contains a substitute clause 47 for certain associations incorporated under CSESocs Incorporation Act 1984.
+	
+#. The financial year of CSESoc is:
+
+   #. the period of time commencing on the date of incorporation of CSESoc and ending on the following 31 December, and
+   #. each period of 12 months after the expiration of the previous financial year of CSESoc, commencing on 1 January and ending on the following 31 December.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:864]:
```suggestion
   #. the period of time commencing on the date of incorporation of CSESoc and ending on the following 30 June, and
   #. each period of 12 months after the expiration of the previous financial year of CSESoc, commencing on 1 July and ending on the following 30 June.
```

The AGM must be held within 6 months of the financial year end. Given that Arc's required AGM period is usually in Term 3, I would recommend using 1 July as the financial year instead.

File: CONSTITUTION.rst [hunk 18]
```diff
@@ -0,0 +1,864 @@
...
+    Part 2 of this Constitution is structured around Part 2 of the Model Constitution, with sections merged
+    or added from the CSESoc Constitution.
+
+Membership Generally
+--------------------
+
+    This section has been entirely taken from sections 2.1 to 2.5 of the CSESoc Constitution.
+    It has been reformatted, but no substantive additions have been made.
+
+#. There shall be two types of membership: *full member*, and *associated member*.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:117]:
```suggestion
#. There shall be two types of membership: *full member*, and *associate member*.
```

File: CONSTITUTION.rst [hunk 19]
```diff
@@ -0,0 +1,864 @@
...
+
+   #. Ensure that through the activities of the Society’s working groups, its aims are being fulfilled;
+   #. Maintain the finances of the society;
+   #. Communicate with the membership of the Society; and
+   #. Maintain and review policies & procedures of the Society, including its Grievance Resolution Policy & Procedure.
+   #. Control and manage the affairs of CSESoc, and
+   #. Exercise all the functions that may be exercised by CSESoc, other than those functions that are required by this constitution to be exercised by a general meeting of members of CSESoc, and
+   #. Have power to perform all the acts and do all things that appear to the Executive to be necessary or desirable for the proper management of the affairs of CSESoc.
+
+#. There is no maximum number of consecutive terms for which a Executive member may hold office.
```
Comment (Qantas94Heavy) [inline CONSTITUTION.rst:241]: Unless I've missed it, the Act requires the constitution to mention how long "the terms of office of the committee members" is. You can just copy/paste in the model constitution wording here:

> Each member of the Executive is, subject to this constitution, to hold office until immediately before the election of Executive members at the annual general meeting next following the date of the member’s election, and is eligible for re-election.

**Review by Qantas94Heavy — COMMENTED** [review]
We're almost there, just one or two more things to fix up. Other than that LGTM :+1:

File: CONSTITUTION.rst (hunk 19 above)
Comment (tfpk) [reply CONSTITUTION.rst]: For this, we specify:
> #. There is no maximum number of consecutive terms for which a Executive member may hold office. 
We also specify that executives are dissolved by handover meeting.
I've added a note that executive members hold office until the executive is dissolved by handover meeting (subject to the other provisions).

**Comment by Qantas94Heavy** [comment]
Looks awesome, great job! :100: 
Thanks a lot to @tfpk and the CSESoc exec for all the work involved to make this important step happen :slightly_smiling_face:

File: CONSTITUTION.rst (hunk 19 above)
Comment (Qantas94Heavy) [reply CONSTITUTION.rst]: Ah right, I've missed that indeed. Thanks :+1:

**Comment by siddhantvirmani** [comment]
Couple of questions:

* Why was the Anti-Discrimination Act removed?
* Why are the list of members not being kept being by the school anymore?

Just doing my job. ;)

**Pull request merged**